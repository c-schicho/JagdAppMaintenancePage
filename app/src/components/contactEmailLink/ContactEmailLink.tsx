import React from 'react';
import texts from '../../texts/texts.json';

export default function ContactEmailLink(): JSX.Element {
    const { email, subject } = texts.contact;
    const href = `mailto:${email}?subject=${subject}`;

    return <a href={href}>{email}</a>;
}
