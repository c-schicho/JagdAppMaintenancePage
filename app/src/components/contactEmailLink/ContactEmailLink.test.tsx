import React from 'react';
import { render, screen } from '@testing-library/react';
import ContactEmailLink from './ContactEmailLink';

describe('<ContactEmailLink>', () => {
    describe('rendering', () => {
        it('should render the email address', () => {
            render(<ContactEmailLink />);

            expect(screen.getByText('kontakt@jagdplaner.at')).toBeVisible();
        });

        it('should match snapshot', () => {
            const { asFragment } = render(<ContactEmailLink />);

            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('behavior', () => {
        it('should open the mail program with the correct email address and subject', () => {
            render(<ContactEmailLink />);

            expect(screen.getByRole('link')).toHaveAttribute(
                'href',
                'mailto:kontakt@jagdplaner.at?subject=Anfrage Jagdplaner'
            );
        });
    });
});
