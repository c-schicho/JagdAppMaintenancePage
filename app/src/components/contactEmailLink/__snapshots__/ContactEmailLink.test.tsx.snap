// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactEmailLink> rendering should match snapshot 1`] = `
<DocumentFragment>
  <a
    href="mailto:kontakt@jagdplaner.at?subject=Anfrage Jagdplaner"
  >
    kontakt@jagdplaner.at
  </a>
</DocumentFragment>
`;
